Shared/Samples/Augment reality to navigate route/AugmentRealityToNavigateRouteView.swift

```diff
@@ -14,245 +14,241 @@
+        MapView(
+            map: model.map,
+            graphicsOverlays: model.graphicsOverlays
+        )
+        .locationDisplay(model.locationDisplay)
+        .onSingleTapGesture { _, mapPoint in
+            tapLocation = mapPoint
+        }
+        .task(id: tapLocation) {
+            guard let tapLocation else { return }
+            
+            do {
+                try await model.addRouteGraphic(for: tapLocation)
+            } catch {
+                self.error = error
             }
-            .toolbar {
-                ToolbarItemGroup(placement: .bottomBar) {
-                    Button("Start") {
-                        isNavigating = true
-                        Task {
-                            do {
-                                try await startNavigation()
-                            } catch {
-                                print("Failed to start navigation.")
-                            }
-                        }
-                    }
-                    .disabled(isNavigating)
-                }
+        }
+        .task {
+            do {
+                try await model.setUp()
+            } catch {
+                self.error = error
+            }
+        }
+        .overlay(alignment: .top) {
+            instructionText
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                toolbarButtons
             }
         }
+        .onAppear {
+            model.statusText = "Tap to place a start point."
+        }
+        .errorAlert(presentingError: $error)
     }
     
-    /// Creates a graphics overlay and adds a graphic (with solid yellow 3D tube symbol)
-    /// to represent the route.
-    private func makeRouteOverlay(routeResult: RouteResult, routeGraphic: Graphic) -> GraphicsOverlay {
-        let graphicsOverlay = GraphicsOverlay()
-        graphicsOverlay.sceneProperties.surfacePlacement = .absolute
-        let strokeSymbolLayer = SolidStrokeSymbolLayer(
-            width: 1.0,
-            color: .yellow,
-            lineStyle3D: .tube
-        )
-        let polylineSymbol = MultilayerPolylineSymbol(symbolLayers: [strokeSymbolLayer])
-        let polylineRenderer = SimpleRenderer(symbol: polylineSymbol)
-        graphicsOverlay.renderer = polylineRenderer
-        
-        if let originalPolyline = routeResult.routes.first?.geometry {
-            addingElevation(3, to: originalPolyline) { polyline in
-                routeGraphic.geometry = polyline
-                graphicsOverlay.addGraphic(routeGraphic)
+    /// The buttons in the bottom toolbar.
+    @ViewBuilder private var toolbarButtons: some View {
+        Spacer()
+        NavigationStack {
+            NavigationLink {
+                ARRouteSceneView(model: model.sceneModel)
+            } label: {
+                Image(systemName: "camera")
+                    .imageScale(.large)
             }
+            .disabled(!model.didSelectRoute)
```

Comment:
```suggestion
        NavigationLink {
            ARRouteSceneView(model: model.sceneModel)
        } label: {
            Image(systemName: "camera")
                .imageScale(.large)
        }
        .disabled(!model.didSelectRoute)
```

Tested this change with v.next since #484 went in and the back button shows up on iPhone and iPad.

Reply:
Thank you. I removed the navigation stack.